### PR TITLE
Add extension as a keyword for Dotty

### PR DESCRIPTION
**Pull request by tgodzik**
Based on https://github.com/lampepfl/dotty/issues/10076#issuecomment-723095528

### CODE REVIEW

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 1]
```diff
@@ -867,7 +867,7 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     def unapply(token: Token): Boolean = {
       token.is[Ident] || token.is[KwIf] || token.is[KwWhile] || token.is[KwFor] ||
       token.is[KwMatch] || token.is[KwTry] || token.is[KwNew] || token.is[KwThis] ||
-      token.is[KwGiven] || token.is[KwVal]
+      token.is[KwGiven] || token.is[KwVal] || token.is[KwExtension]
       /* extension is ommited here as it is SoftKeyword <-> Ident */
```
Comment (kpbochenek) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:871]: comment can be removed :smile:

File: scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyToken.scala [hunk 2]
```diff
@@ -67,6 +67,7 @@ object LegacyToken {
   final val TYPE = 70
   final val FORSOME = 71
   final val DEF = 72
+  final val EXTENSION = 139
```
Comment (kpbochenek) [inline scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyToken.scala:70]: can we move it to below `GIVEN = 76` and assign number 77?
or below `CTXARROW = 138` 

just to keep this list sorted

**Review by kpbochenek — APPROVED** [review]
scalameta ahead of compiler changes :P 

(just minor cosmetic comments)

LGTM! :+1:

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala (hunk 1 above)
Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala]: Removed

File: scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyToken.scala (hunk 2 above)
Comment (tgodzik) [reply scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyToken.scala]: Changed it to 76, good point.

**Pull request merged**